File: db4on/core/src/core/com/db4o/MWriteUpdate.cs
namespace com.db4o
{
	internal sealed class MWriteUpdate : com.db4o.MsgObject
	{
		internal sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 sock)
		{
			int yapClassId = _payLoad.ReadInt();
			com.db4o.YapFile stream = (com.db4o.YapFile)GetStream();
			Unmarshall(com.db4o.YapConst.INT_LENGTH);
			lock (stream.i_lock)
			{
				com.db4o.YapClass yc = stream.GetYapClass(yapClassId);
				_payLoad.WriteEmbedded();
				com.db4o.inside.slots.Slot oldSlot = _trans.GetCommittedSlotOfID(_payLoad.GetID()
					);
				stream.GetSlotForUpdate(_payLoad);
				yc.AddFieldIndices(_payLoad, oldSlot);
				_payLoad.WriteEncrypt();
			}
			return true;
		}
	}
}

File: db4oj/core/src/com/db4o/internal/cs/messages/MWriteUpdate.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.internal.cs.messages;

import com.db4o.internal.*;
import com.db4o.internal.slots.*;

public final class MWriteUpdate extends MsgObject implements ServerSideMessage {
	
	public final boolean processAtServer() {
	    int yapClassId = _payLoad.readInt();
	    LocalObjectContainer stream = (LocalObjectContainer)stream();
	    unmarshall(_payLoad._offset);
	    synchronized(streamLock()){
	        ClassMetadata yc = stream.classMetadataForId(yapClassId);
			int id = _payLoad.getID();
			transaction().dontDelete(id);
            Slot oldSlot = ((LocalTransaction)transaction()).getCommittedSlotOfID(id);
            stream.getSlotForUpdate(_payLoad);
			yc.addFieldIndices(_payLoad, oldSlot);
            _payLoad.writeEncrypt();
		}
		return true;
	}
}